libbeat/common/jsontransform/jsonhelper.go

```diff
@@ -110,3 +115,23 @@ func removeKeys(keys map[string]interface{}, names ...string) {
 		delete(keys, name)
 	}
 }
+
+func parseTimestamp(timestamp string) (time.Time, error) {
+	validFormats := []string{
+		time.RFC3339,
+		iso8601,
+	}
+
+	var err error
+	for _, f := range validFormats {
+		ts, parseErr := time.Parse(f, timestamp)
+		if parseErr != nil {
+			err = multierror.Append(err, parseErr)
+			continue
+		}
+
+		return ts, nil
+	}
+
+	return time.Now(), nil
```

Comment:
returning `time.Now()` might be a semantic/breaking change for some users. The `@timestamp` was already set and would have been kept if the parse failed. Returning `time.Now()` here means that we will now overwrite existing timestamps no matter what.